memote/suite/collect.py

```diff
@@ -109,6 +110,56 @@ def _collect_git_info(self):
                 " ")
             self._meta["commit_hash"] = self.commit.hexsha
 
+    def pytest_runtest_protocol(self, item):
+        """
+        Collect test function docstring for tooltip text in the reports.
+
+        Parameters
+        ----------
+        item : pytest.Item
+            A test item objects, i.e. a test function found by pytest.
+
+        """
+        module_name = item.obj.__module__
+        item_name = str(item.name).lower()
+
+        if module_name not in self._data:
+            self._data[module_name] = dict()
+
+        if item_name not in self._data[module_name]:
+            self._data[module_name][item_name] = dict()
+
+        if item.obj.__doc__:
+            docstring = " ".join(FunctionDoc(item.obj)["Summary"])
+            self._data[module_name][item_name]["summary"] = docstring
+        else:
+            self._data[module_name][item_name]["summary"] = "No summary " \
+                                                            "available."
+
+    def pytest_report_teststatus(self, report):
+        """
+        Log pytest results for each test.
+
+        The categories are passed, failed, error, skipped and marked to fail.
+
+        Parameters
+        ----------
+        report : TestReport
+            A test report object from pytest with test case result.
+
+        """
+        module_name = report.location[0].split("/")[-1].split(".")[0]
+        item_name = str(report.location[-1]).lower()
+
+        if module_name not in self._data:
+            self._data[module_name] = dict()
+
+        if item_name not in self._data[module_name]:
+            self._data[module_name][item_name] = dict()
+
+        self._data[module_name][item_name]["duration"] = report.duration
+        self._data[module_name][item_name]["result"] = report.outcome
```

Comment:
I think you have to be careful with the report outcome. If it was an error then simply returning it will raise the error at this point (if I remember corrrectly). Please make sure this isn't the case or handle appropriately.